test/models/submission/CharityTransformerSpec.scala

```diff
@@ -458,7 +458,123 @@ class CharityTransformerSpec extends SpecBase with CharityTransformerTodoPages {
+            |			"northernIreland": true,
+            |			"ukWide": true,
+            |			"overseas": false
+            |		}
+            |}""".stripMargin
+
+        localUserAnswers.data.transform(jsonTransformer.userAnswersToOperationAndFunds).asOpt.value mustBe Json.parse(expectedJson)
+      }
+
+      "convert the correct OperationAndFunds object with England, Scotland and Northern Ireland" in {
+
+        val localUserAnswers =  emptyUserAnswers.set(AccountingPeriodEndDatePage,
+          MonthDay.fromDateFields(new JLocalDate(2020, 1, 1).toDate))(MongoDateTimeFormats.localDayMonthWrite).flatMap(
+          _.set(IsFinancialAccountsPage, true)).flatMap(
+          _.set(FundRaisingPage, FundRaisingOptions.values.toSet)).flatMap(
+          _.set(OperatingLocationPage, Set[OperatingLocationOptions](OperatingLocationOptions.England, OperatingLocationOptions.Scotland, OperatingLocationOptions.NorthernIreland))
+        ).success.value
+
+        val expectedJson =
+          """{
+            |		"operationAndFunds": {
+            |			"operationAndFundsCommon": {
+            |       "accountPeriodEnd": "0101",
+            |       "financialAccounts": true
+            |			},
+            |			"futureFunds": "other, donations, tradingSubsidiaries, tradingIncome, fundraising, grants, membershipSubscriptions, investmentIncome",
+            |			"otherAreaOperation": true,
+            |			"englandAndWales": true,
+            |			"scotland": true,
+            |			"northernIreland": true,
+            |			"ukWide": true,
+            |			"overseas": false
+            |		}
+            |}""".stripMargin
+
+        localUserAnswers.data.transform(jsonTransformer.userAnswersToOperationAndFunds).asOpt.value mustBe Json.parse(expectedJson)
+      }
+
+      "convert the correct OperationAndFunds object with England and Northern Ireland" in {
+
+        val localUserAnswers =  emptyUserAnswers.set(AccountingPeriodEndDatePage,
+          MonthDay.fromDateFields(new JLocalDate(2020, 1, 1).toDate))(MongoDateTimeFormats.localDayMonthWrite).flatMap(
+          _.set(IsFinancialAccountsPage, true)).flatMap(
+          _.set(FundRaisingPage, FundRaisingOptions.values.toSet)).flatMap(
+          _.set(OperatingLocationPage, Set[OperatingLocationOptions](OperatingLocationOptions.England, OperatingLocationOptions.NorthernIreland))
+        ).success.value
+
+        val expectedJson =
+          """{
+            |		"operationAndFunds": {
+            |			"operationAndFundsCommon": {
+            |       "accountPeriodEnd": "0101",
+            |       "financialAccounts": true
+            |			},
+            |			"futureFunds": "other, donations, tradingSubsidiaries, tradingIncome, fundraising, grants, membershipSubscriptions, investmentIncome",
+            |			"otherAreaOperation": true,
+            |			"englandAndWales": true,
+            |			"scotland": false,
+            |			"northernIreland": true,
+            |			"ukWide": false,
```

Comment:
below condition is not working, please have a look
England + Wales + Scotland + Northern Ireland  =>  ukWide (true)
England + Scotland + Northern Ireland  =>  ukWide (false)
Wales+ Scotland + Northern Ireland  =>  ukWide (false)